src/github_runner_image_builder/builder.py

```diff
@@ -0,0 +1,609 @@
+            services.
+    """
+    try:
+        # use subprocess run rather than operator-libs-linux's systemd library since the library
+        # does not provide full features like mask.
+        subprocess.run(
+            ["/usr/bin/systemctl", "stop", APT_TIMER], check=True, timeout=30
+        )  # nosec: B603
+        subprocess.run(
+            ["/usr/bin/systemctl", "disable", APT_TIMER], check=True, timeout=30
+        )  # nosec: B603
+        subprocess.run(
+            ["/usr/bin/systemctl", "mask", APT_SVC], check=True, timeout=30
+        )  # nosec: B603
+        subprocess.run(
+            ["/usr/bin/systemctl", "stop", APT_UPGRADE_TIMER], check=True, timeout=30
+        )  # nosec: B603
+        subprocess.run(  # nosec: B603
+            ["/usr/bin/systemctl", "disable", APT_UPGRADE_TIMER], check=True, timeout=30
+        )
+        subprocess.run(
+            ["/usr/bin/systemctl", "mask", APT_UPGRAD_SVC], check=True, timeout=30
+        )  # nosec: B603
+        subprocess.run(
+            ["/usr/bin/systemctl", "daemon-reload"], check=True, timeout=30
+        )  # nosec: B603
+        subprocess.run(  # nosec: B603
+            ["/usr/bin/apt-get", "remove", "-y", "unattended-upgrades"], check=True, timeout=30
+        )
+    except subprocess.SubprocessError as exc:
+        raise UnattendedUpgradeDisableError from exc
+
+
+class SystemUserConfigurationError(Exception):
+    """Represents an error while adding user to chroot env."""
+
+
+UBUNTU_USER = "ubuntu"
+DOCKER_GROUP = "docker"
+MICROK8S_GROUP = "microk8s"
+LXD_GROUP = "lxd"
+UBUNTU_HOME = Path("/home/ubuntu")
+
+
+def _configure_system_users() -> None:
+    """Configure system users.
+
+    Raises:
+        SystemUserConfigurationError: If there was an error configuring ubuntu user.
+    """
+    try:
+        subprocess.run(  # nosec: B603
+            ["/usr/sbin/useradd", "-m", UBUNTU_USER], check=True, timeout=30
+        )
+        with (UBUNTU_HOME / ".profile").open("a") as profile_file:
+            profile_file.write("PATH=$PATH:/home/ubuntu/.local/bin\n")
+        with (UBUNTU_HOME / ".bashrc").open("a") as bashrc_file:
+            bashrc_file.write("PATH=$PATH:/home/ubuntu/.local/bin\n")
+        subprocess.run(  # nosec: B603
+            ["/usr/sbin/groupadd", MICROK8S_GROUP], check=True, timeout=30
+        )
+        subprocess.run(  # nosec: B603
+            ["/usr/sbin/usermod", "-aG", DOCKER_GROUP, UBUNTU_USER], check=True, timeout=30
+        )
+        subprocess.run(  # nosec: B603
+            ["/usr/sbin/usermod", "-aG", MICROK8S_GROUP, UBUNTU_USER], check=True, timeout=30
+        )
+        subprocess.run(  # nosec: B603
+            ["/usr/sbin/usermod", "-aG", LXD_GROUP, UBUNTU_USER], check=True, timeout=30
+        )
+        subprocess.run(  # nosec: B603
+            ["/usr/bin/chmod", "777", "/usr/local/bin"], check=True, timeout=30
```

Comment:
Why are we doing this ?

Reply:
This is to allow runners to allow installing executables. @yhaliaw I think this is to also mimic the behaviors of the github runners, would it be correct?